lib/ansible/galaxy/collection.py

```diff
@@ -220,16 +236,16 @@ def from_tar(path, validate_certs, parent=None):
         name = meta['name']
         version = meta['version']
 
-        return CollectionRequirement(namespace, name, path, None, [version], version, parent=parent,
+        return CollectionRequirement(namespace, name, path, None, [version], version, force, parent=parent,
                                      validate_certs=validate_certs, metadata=meta, files=files)
 
     @staticmethod
-    def from_path(path, validate_certs, parent=None):
+    def from_path(path, validate_certs, force, parent=None):
         info = {}
         for file_name, property_name in CollectionRequirement._FILE_MAPPING:
             file_path = os.path.join(path, file_name)
             if os.path.exists(file_path):
-                with open(file_path, 'rb') as file_obj:
+                with open(file_path, 'r') as file_obj:
```

Comment:
I'm pretty sure that @abadger advised against loading JSON from unicode text because it's faulty under Python 2.

Reply:
Will see what he has to say but you cannot load from bytes on 3.5 hence why I removed the `b`.

Reply:
Yeah, he also said to use `loads`, not `load`. I remember debugging its internals and it was processing the stream in chunks internally and turned some of them into unicode while keeping the others in bytes and then crashed trying to combine those...

Reply:
Ah... It was about dumping: https://github.com/ansible/ansibullbot/pull/1078#issuecomment-430394122. I was creating some shims when making the ansibullbot repo py3-ready.

Reply:
Interesting, I've tested this with BytesIO and it works...